kaal-infrastructure/src/main/kotlin/cz/eman/kaal/infrastructure/api/retrofit/KaalRetrofitCaller.kt

```diff
@@ -2,127 +2,207 @@ package cz.eman.kaal.infrastructure.api.retrofit
-     * @param map function mapping [Dto] object to [Unit] object (kept to be used for logging or other functions)
-     * @return [Result] with [Unit]
-     * @since 0.9.0
-     */
-    suspend fun <Dto> callResultUnit(
-        responseCall: suspend () -> Response<Dto>,
-        errorMessage: () -> String?,
-        map: suspend (Dto) -> Unit = { }
-    ): Result<Unit> = callResultNull(responseCall, errorMessage, map).map { }
+    ): Result<Optional<T>> {
+        return callResultNull(responseCall, errorMessage, map).map { data ->
+            if (data != null) {
+                Optional.of(data)
+            } else {
+                Optional.empty()
+            }
+        }
+    }
 
     /**
      * Calls a [responseCall] and handles the result by mapping the [Dto] object to nullable [T] on success else it
      * returns [Result.Error] on any error or exception. Optional allows support for empty body.
      *
+     * Handles response using [handleResponse] and is returned. If any exception occurs it is handled by
+     * [handleCallException].
+     *
      * @param responseCall Retrofit2 call to handle
      * @param errorMessage used to modify error message
      * @param map function mapping [Dto] object to [T] object
      * @return [Result] with [T]? (nullable) allowing success with empty body
+     * @see responseCall
+     * @see handleResponse
+     * @see handleCallException
      * @since 0.9.0
      */
     suspend fun <Dto, T> callResultNull(
         responseCall: suspend () -> Response<Dto>,
         errorMessage: () -> String?,
         map: suspend (Dto) -> T
-    ): Result<T?>
+    ): Result<T?> {
+        return try {
+            handleResponse(responseCall(), map)
+        } catch (ex: Exception) {
+            handleCallException(ex, errorMessage())
+        }
+    }
 
     /**
      * Calls a [responseCall] and handles the result by mapping the [Dto] object to [Optional]<[T]> on success else it
      * returns [Result.Error] on any error or exception. Optional allows support for empty body. Result of this function
      * is a [Pair] of [Result] and [Response] so you can easily access any response metadata you need (like headers,
      * result code and others provided by Retrofit2).
      *
+     * This function calls [callResultResponseNull] which can return [Result] result with null body. This result is then
+     * mapped to [Optional] using [Result.map].
+     *
      * @param responseCall Retrofit2 call to handle
      * @param errorMessage used to modify error message
      * @param map function mapping [Dto] object to [T] object
-     * @return [Pair] of [Result] with [T] data to [Response] with [Dto] object
+     * @return [Pair] of [Result] with [T] data to [Response]? (nullable) with [Dto] object
+     * @see responseCall
+     * @see handleResponse
+     * @see handleCallException
      * @since 0.9.0
      */
     @RequiresApi(Build.VERSION_CODES.N)
     suspend fun <Dto, T> callResultResponseOptional(
```

Comment:
This special case should be handled outside of `KaalRetrofitCaller`, because user / developer can ignore `Result<...>` and work only with `Response<...>`

Reply:
As we discussed I would like to keep it for now but we can create an issue to discuss this further.

Reply:
OK 👍